src/components/ArtifactDataTable.vue

```diff
@@ -22,30 +28,62 @@
   import { SearchInput } from '@/components'
   import { localization } from '@/localization'
   import { TableArtifact } from '@/models'
+  import { isTableArtifactData, isArrayOfMaps, isMapOfArrays } from '@/types/artifact'
+  import { parseUnknownJson } from '@/utilities/json'
 
   const props = defineProps<{
     artifact: TableArtifact,
   }>()
 
-  const data = computed<TableData[]>(() => props.artifact.data)
+  const data = computed(() => parseUnknownJson(props.artifact.data))
+  const isValid = computed(() => isTableArtifactData(props.artifact.data))
+
+  const normalizedData = computed(() => {
+    if (isArrayOfMaps(data.value)) {
+      return data.value
+    }
+
+    if (isMapOfArrays(data.value)) {
```

Comment:
Can give me a small example of this? Just to visualize

Reply:
Yup!

This:
```json
{
  "id": [0, 1, 2, 3, 4, 5, 6, 7],
  "name": ["Dublin", "London", "New York", "Oslo", "Paris", "Rome", "Stockholm", "Tokyo"],
  "lat": [53.3498, 51.5074, 40.7128, 59.9139, 48.8566, 41.9028, 59.3293, 35.6762],
  "lon": [-6.2603, -0.1278, -74.0060, 10.7522, 2.3522, 12.4964, 18.0686, 139.6503],
}
```

vs:
```json
[
    {
        "id": 0, "name": "Dublin", "lat": 53.3498, "lon": -6.2603,
    },
    {
        "id": 1, "name": "London", "lat": 51.5074, "lon": -0.1278,
    },
    {
        "id": 2, "name": "New York", "lat": 40.7128, "lon": -74.0060,
    },
    {
        "id": 3, "name": "Oslo", "lat": 59.9139, "lon": 10.7522,
    },
    {
        "id": 4, "name": "Paris", "lat": 48.8566, "lon": 2.3522,
    },
    {
        "id": 5, "name": "Rome", "lat": 41.9028, "lon": 12.4964,
    },
    {
        "id": 6, "name": "Tokyo", "lat": 35.6895, "lon": 139.6917,
    },
    {
        "id": 7, "name": "Vancouver", "lat": 49.2827, "lon": -123.1207,
    }
]
```

Reply:
Thank you!